test/test_main_functionality.py

```diff
@@ -0,0 +1,79 @@
+import unittest
```

Comment:
Nitpick: Is it really necessary to test `main.py`? Because the methods you are using in `main.py` have been tested in their individual classes.